org.jacoco.core/src/org/jacoco/core/tools/ExecDumpClient.java

```diff
@@ -123,7 +123,10 @@ public ExecFileLoader dump(final InetAddress address, final int port)
 					.setExecutionDataVisitor(loader.getExecutionDataStore());
 
 			remoteWriter.visitDumpCommand(dump, reset);
-			remoteReader.read();
+
+			if (!remoteReader.read()) {
+				throw new IOException("Socket closed unexpectedly.");
```

Comment:
@marchof seems that this change is a bit unrelated to PR and test is missing for this case